### PR TITLE
mcs: extract the common part as base server

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

There is too many duplicated code in the different services. Maintaining one copy will help us reduce the probability of making mistakes.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/server/server.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		startTimestamp: time.Now().Unix(),
+	}
+}
+
+// Context returns the context of server.
+func (bs *BaseServer) Context() context.Context {
+	return bs.ctx
+}
+
+// GetDelegateClient returns grpc client connection talking to the forwarded host
```
Comment (lhy1024) [inline pkg/mcs/server/server.go:64]:
```suggestion
// GetDelegateClient returns grpc client connection talking to the forwarded host.
```

**Review by lhy1024 — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

File: pkg/mcs/server/server.go (hunk 1 above)
Comment (rleungx) [reply pkg/mcs/server/server.go:64]: Maybe in another PR.

**Pull request merged**